core/templates/dev/head/app.js

```diff
@@ -1220,3 +1220,20 @@ oppia.factory('codeNormalizationService', [function() {
     }
   };
 }]);
+
+// Service for retrieving algorithm id of classifier for a given interaction id.
+oppia.factory('algorithmIdRetrievalService', [function() {
```

Comment:
Not sure why you need this factory. Store interactionClassifierMapping in assets/constants.js and remove it from feconf if it's meant to be the same constant in both the frontend and the backend.